buffer/src/main/java/io/netty/buffer/ReadOnlyByteBuf.java

```diff
@@ -40,7 +40,7 @@
     private final ByteBuf buffer;
 
     public ReadOnlyByteBuf(ByteBuf buffer) {
-        super(buffer.maxCapacity());
+        super(buffer.capacity()); // ReadOnly buffers are not allowed to grow capacity.
```

Comment:
Note this is still inconsistent with `ReadOnlyByteBufferBuf` which uses `buffer.remaining()`, but it seems more correct than what is there now because in general because `ReadOnlyByteBuf`'s capacity cannot be adjusted and no bytes can be written to it.

Reply:
from @njhill 
> whether these changes are safe/appropriate to make to ReadOnlyByteBuf

Reply:
As a read-only view of an existing buffer, some users might expect to be able to still read past the initial capacity if the source buffer's capacity increases after this was created.

This buffer would end up in a kind of weird state where `buf.capacity() > buf.maxCapacity()`...

Reply:
I have not thought through all of the implications of this but maybe having instead overriding `capacity()` to return `buffer.writerIndex()` might make sense.

Reply:
@Scottmitch I agree with @njhill if the underlying buffer expands the user should be be able to access the bytes via the read-only view.

Reply:
> This buffer would end up in a kind of weird state where buf.capacity() > buf.maxCapacity()...

Agreed this would be wrong ... let me revert this part of the change.

Reply:
@Scottmitch can we also add a test that verifies that "use-case" ?